xchange-stream-binance/src/main/java/info/bitrich/xchangestream/binance/BinanceStreamingMarketDataService.java

```diff
@@ -356,30 +342,67 @@ private OrderbookSubscription connectOrderBook(CurrencyPair currencyPair) {
     return observable;
   }
 
-  private BinanceWebsocketTransaction<TickerBinanceWebsocketTransaction> tickerTransaction(
-      JsonNode node) {
+  private BinanceWebsocketTransaction<TickerBinanceWebsocketTransaction> tickerTransaction(JsonNode node) {
     try {
       return mapper.readValue(mapper.treeAsTokens(node), TICKER_TYPE);
     } catch (IOException e) {
       throw new ExchangeException("Unable to parse ticker transaction", e);
     }
   }
 
-  private BinanceWebsocketTransaction<DepthBinanceWebSocketTransaction> depthTransaction(
-      JsonNode node) {
+  private BinanceWebsocketTransaction<DepthBinanceWebSocketTransaction> depthTransaction(JsonNode node) {
     try {
       return mapper.readValue(mapper.treeAsTokens(node), DEPTH_TYPE);
     } catch (IOException e) {
       throw new ExchangeException("Unable to parse order book transaction", e);
     }
   }
 
-  private BinanceWebsocketTransaction<TradeBinanceWebsocketTransaction> tradeTransaction(
-      JsonNode node) {
+  private BinanceWebsocketTransaction<TradeBinanceWebsocketTransaction> tradeTransaction(JsonNode node) {
     try {
       return mapper.readValue(mapper.treeAsTokens(node), TRADE_TYPE);
     } catch (IOException e) {
       throw new ExchangeException("Unable to parse trade transaction", e);
     }
   }
+
+  private Stream<OrderBookUpdate> extractOrderBookUpdates(CurrencyPair currencyPair,
+          DepthBinanceWebSocketTransaction depthTransaction) {
+    BinanceOrderbook orderBookDiff = depthTransaction.getOrderBook();
+
+    Stream<OrderBookUpdate> bidStream = orderBookDiff.bids.entrySet().stream()
```

Comment:
a little cleaner then this is to use .foreach(k,v -> {})

Reply:
Agree, but in this case i am returning a stream.

Reply:
I see you're right